### ISSUE TITLE
[BUG] Does not work when libvirt's split daemon mode is enabled

### ISSUE THREAD

**Issue by andreabolognani** [code-ready/crc#1090]
### General information

  * OS: Linux
  * Hypervisor: KVM
  * Did you run `crc setup` before starting it? Yes

## CRC version
```bash
$ crc version
crc version: 1.7.0+fa7e558
OpenShift version: 4.3.1 (embedded in binary)
$
```

## CRC status

N/A

## CRC config
```bash
$ crc config view
$
```

## Host Operating System
```bash
$ cat /etc/os-release
NAME=Fedora
VERSION="31 (Workstation Edition)"
ID=fedora
VERSION_ID=31
VERSION_CODENAME=""
PLATFORM_ID="platform:f31"
PRETTY_NAME="Fedora 31 (Workstation Edition)"
ANSI_COLOR="0;34"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:31"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f31/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=31
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=31
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation
$
```

### Steps to reproduce

When split daemon mode is enabled, `crc start` fails:

```
$ crc start --log-level debug
DEBU No new version available. The latest version is 1.7.0
INFO Checking if oc binary is cached
DEBU oc binary already cached
INFO Checking if podman remote binary is cached
DEBU podman remote binary already cached
INFO Checking if running as non-root
INFO Checking if Virtualization is enabled
DEBU Checking if the vmx/svm flags are present in /proc/cpuinfo
DEBU CPU virtualization flags are good
INFO Checking if KVM is enabled
DEBU Checking if /dev/kvm exists
DEBU /dev/kvm was found
INFO Checking if libvirt is installed
DEBU Checking if 'virsh' is available
DEBU 'virsh' was found in /usr/bin/virsh
INFO Checking if user is part of libvirt group
DEBU Checking if current user is part of the libvirt group
DEBU Running '/usr/bin/groups abologna'
DEBU Current user is already in the libvirt group
INFO Checking if libvirt is enabled
DEBU Checking if libvirtd.service is enabled
DEBU Running '/usr/bin/systemctl is-enabled libvirtd'
DEBU Command failed: exit status 1
DEBU stdout: disabled
DEBU stderr:
DEBU Error checking if libvirtd service is enabled
FATA Error checking if libvirtd service is enabled
$
```

On my system, while `libvirtd` is indeed disabled, `virtqemud` is enabled:

```
$ systemctl is-enabled libvirtd
disabled
$ systemctl is-enabled virtqemud
enabled
$
```

and all the usual libvirt operations work just fine:

```
$ virsh -c qemu:///system net-list
 Name      State    Autostart   Persistent
--------------------------------------------
 crc       active   yes         yes

$ virsh -c qemu:///system list --all
 Id   Name      State
--------------------------
 -    crc       shut off

$ virsh -c qemu:///system dumpxml crc | wc -l
75
$
```

Additional information on split daemon mode, including the steps necessary to enable it, can be found on [the libvirt website](https://libvirt.org/daemons.html).

Split daemon mode has been introduced in [libvirt 5.7.0](https://libvirt.org/news.html#v5.7.0) and, while it's still considered experimental, it's solid enough for regular use and will eventually be the recommended way to deploy libvirt.

### Expected

CRC works when split daemon mode is enabled.

### Actual

CRC doesn't work when split daemon mode is enabled.

### Logs

See above.

**Comment by gbraad** [comment]
Currently not supported.
For some background: https://www.berrange.com/posts/2020/02/04/libvirt-split-of-the-monolithic-libvirtd-daemon/

@zeenix could you have a look at what might need changes on our end? Just collect info here as a spike. No need to implement yet.

**Comment by zeenix** [comment]
@gbraad sure thing.

**Comment by cfergeau** [comment]
Some advice from IRC
```
17:51 < danpb> abologna: if  libvirtd.socket or virtqemud.socket units exist, then they can skip any logic for manually starting services
17:51 < danpb> abologna: if neither of those units exist, then it is safe to assume split daemons don't exist, and so starting libvirtd is safe
```

**Comment by andreabolognani** [comment]
Right, I was just about to comment this additional information: if either `libvirtd.socket` or `virtqemud.socket` are present, they should be considered equivalent to the corresponding `.service` for the purpose of CRC's checks: it's a perfectly valid configuration, for example, to leave `libvirtd.service` as disabled and `libvirtd.socket` as enabled and rely on socket activation.

**Comment by andreabolognani** [comment]
As suggested by @cfergeau elsewhere, running

```
$ crc config set skip-check-libvirt-enabled true
$ crc config set skip-check-libvirt-running true
```

is an effective workaround for the time being.

**Comment by gbraad** [comment]
This is at the moment not generally available, but what is the expectance of rolling this out in for instance Fedora?

**Comment by andreabolognani** [comment]
> This is at the moment not generally available, but what is the expectance of rolling this out in for instance Fedora?

The libvirt packages in Fedora 32 will contain all the necessary bits for split daemon mode to work; it will, however, be an opt-in configuration rather than the default one.

**Comment by andreabolognani** [comment]
@cfergeau As far as I can tell, CRC will still not work out of the box when split daemon mode is enabled, right? If so, the issue should remain open.

**Comment by cfergeau** [comment]
This was fixed by https://github.com/code-ready/crc/pull/1302 which should be in the latest release. I expected this to be closed automatically when PR #1302 got merged.

**Comment by cfergeau** [comment]
https://github.com/code-ready/crc/blob/master/pkg/crc/preflight/preflight_checks_linux.go#L187-L194 should do the trick for checking for split libvirtd setups

**Comment by andreabolognani** [comment]
> This was fixed by #1302 which should be in the latest release. I expected this to be closed automatically when PR #1302 got merged.

I just tried CRC 1.13.0 and indeed it worked with split daemon mode without further tweaking! Thanks a lot for taking care of this :)
